lib/axon/compiler.ex

```diff
@@ -145,7 +145,17 @@ defmodule Axon.Compiler do
        ) do
     op_counts = Map.update(op_counts, :constant, 1, fn x -> x + 1 end)
 
-    tensor = Nx.backend_copy(tensor, Nx.Defn.Expr)
+    tensor =
+      tensor =
+      case tensor do
+        %Nx.Tensor{data: %Nx.Defn.Expr{}} ->
+          tensor
+
+        t ->
+          Nx.backend_copy(t, Nx.Defn.Expr)
+      end
```

Comment:
We should fix this in Nx. A backend copy of Nx.Defn.Expr to Nx.Defn.Expr should return itself.